src/backend/distributed/utils/foreign_key_relationship.c

```diff
@@ -284,6 +289,60 @@ GetConnectedListHelper(ForeignConstraintRelationshipNode *node, List **adjacentN
 }
 
 
+/*
+ * GetNeighbourList returns copy of relevant adjacency list of given
+ * ForeignConstraintRelationshipNode object depending on the isReferencing
+ * flag.
+ */
+static List *
+GetNeighbourList(ForeignConstraintRelationshipNode *relationshipNode, bool isReferencing)
+{
+	if (isReferencing)
+	{
+		return list_copy(relationshipNode->backAdjacencyList);
```

Comment:
previously we were using the value directly. Is that a bug that is being fixed here?

Reply:
no, I just wanted to make sure that we are on the safe side for further improvements that would be done in #4439

Reply:
but I can revert this change if it doesn't make much sense

Reply:
Maybe revert it, since shallow copies are a bit confusing.